### ISSUE TITLE
Build with custom cuda/cudnn version

### ISSUE THREAD

**Issue by Cospel** [tensorflow/addons#770]
**Describe the feature and the current behaviour/state.**

Right now the build is locked to cuda 10.1. However tensorflow2+ is able to build from source with different CUDA versions (by specifying it through configure). Can this package mimic the tf2 build process?

For example I'm working mostly with docker containers and server instances which has cuda 10.0 and doing custom builds of tf2 for this cuda version.

Right now we are unable to install or build addons as it is locked to 10.1. It will be great if users are able to set during the configuration their own cuda version as it is in tensorflow2+.

This would also help when future versions of cuda are available and we want to test them.

**Relevant information**
- Are you willing to contribute it (yes/no): -
- Are you willing to maintain it going forward? (yes/no): - 
- Is there a relevant academic paper? (if so, where): -
- Is there already an implementation in another framework? (if so, where):
- Was it part of tf.contrib? (if so, where): -

**Which API type would this fall under (layer, metric, optimizer, etc.)**

**Who will benefit with this feature?**
Everyone.

**Any other info.**

**Comment by seanpmorgan** [comment]
> **Describe the feature and the current behaviour/state.**
> 
> Right now the build is locked to cuda 10.1. However tensorflow2+ is able to build from source with different CUDA versions (by specifying it through configure). Can this package mimic the tf2 build process?
> 
> For example I'm working mostly with docker containers and server instances which has cuda 10.0 and doing custom builds of tf2 for this cuda version.
> 
> Right now we are unable to install or build addons as it is locked to 10.1. It will be great if users are able to set during the configuration their own cuda version as it is in tensorflow2+.
> 
> This would also help when future versions of cuda are available and we want to test them.
> 
> **Relevant information**
> 
> * Are you willing to contribute it (yes/no): -
> * Are you willing to maintain it going forward? (yes/no): -
> * Is there a relevant academic paper? (if so, where): -
> * Is there already an implementation in another framework? (if so, where):
> * Was it part of tf.contrib? (if so, where): -
> 
> **Which API type would this fall under (layer, metric, optimizer, etc.)**
> 
> **Who will benefit with this feature?**
> Everyone.
> 
> **Any other info.**

Hi @Cospel! Thanks for bringing this up. So we actually use the [same script for identifying CUDA](https://github.com/tensorflow/addons/blob/master/build_deps/toolchains/gpu/find_cuda_config.py) as TF-Core. This means it should be pretty straight forward to enable this.

We're in the middle of some pretty big BUILD revamping (enable Windows, etc.) so this might get pushed a week or two, but we'll be sure to support this within the config script.

**Comment by SmileTM** [comment]
have a good news to you.
I have successed to bazel and install the addons in cuda 9.0 and cudnn 7.
You can refer
http://s-tm.cn/2019/12/19/tensorflow-addons/

**Comment by seanpmorgan** [comment]
This should be significantly easier to manage if we can use an exported TF method for finding CUDA. This looks possible if https://github.com/tensorflow/tensorflow/pull/38964 merges

**Comment by Cospel** [comment]
Thanks for the great news! Let's hope that it will be merged soon 👍

**Comment by Cospel** [comment]
After a lot of tries and nights, we were able to build tf-addons without no segmentation fault #1298 #1277.

Here is what I did: https://gist.github.com/Cospel/fb9c313cdb83d5e474aa0e3f956d14e0

Here is the built pip for tf-addons on cuda10, cuddn7.5 for centos and python3.6 and tf2.2: https://github.com/Ximilar-com/tf2-wheels/blob/master/README.md#tf2-addons

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
